src/shared/copy.c

```diff
@@ -1455,6 +1471,26 @@ int copy_file_atomic_at_full(
         assert(to);
         assert(!FLAGS_SET(copy_flags, COPY_LOCK_BSD));
 
+        if (copy_flags & COPY_REPLACE_UPDATED) {
+                struct stat buf;
+                nsec_t orig_ts, replace_ts;
+
+                if (fstatat(dir_fdf, from, &buf, AT_SYMLINK_NOFOLLOW) < 0)
+                        return -errno;
+                replace_ts = timespec_load_nsec(&buf.st_mtim);
+
+                if (fstatat(dir_fdt, to, &buf, AT_SYMLINK_NOFOLLOW) < 0)
+                        return -errno;
+                orig_ts = timespec_load_nsec(&buf.st_mtim);
+
+                assert(orig_ts != NSEC_INFINITY);
+                assert(replace_ts != NSEC_INFINITY);
```

Comment:
assertions for this? I don't think we can assume that the timestamps set are always valid. for example fuse sets the timestamps to -1 iirc in some cases. I think we must be prepared here for totally invalid timestamps.

hence, here too, stat_inode_unmodified() seems like the way to go